perception/optimization_based_point_cloud_registration.py

```diff
@@ -0,0 +1,243 @@
+import numpy as np
+import meshcat.geometry as g
+import meshcat.transformations as tf
+from scipy.optimize import fmin_bfgs
+from sklearn.neighbors import NearestNeighbors
+
+
+def MakeMeshcatColorArray(N, r, g, b):
```

Comment:
A similar function is defined in `iterative_closest_point.py`. It can be moved into a separate file and shared by different scripts.

Reply:
I'll put in a TODO for now and I'll move it when I get visualization stuff working.